base/set.jl

```diff
@@ -178,6 +178,49 @@ function unique(f, C)
     out
 end
 
+"""
+    unique!(f, A::AbstractVector)
+
+Selects one value from `A` for each unique value produced by `f` applied to
+elements of `A` , then return the modified A.
+
+# Examples
+```jldoctest
+julia> unique!(x -> x^2, [1, -1, 3, -3, 4])
+3-element Array{Int64,1}:
+ 1
+ 3
+ 4
+
+julia> unique!(n -> n%3, [5, 1, 8, 9, 3, 4, 10, 7, 2, 6])
+3-element Array{Int64,1}:
+ 5
+ 1
+ 9
+
+julia> unique!(iseven, [2, 3, 5, 7, 9])
+2-element Array{Int64,1}:
+ 2
+ 3
+```
+"""
+function unique!(f, A::AbstractVector)
+    seen = Set{eltype(A)}()
+    idxs = eachindex(A)
+    y = iterate(idxs)
+    count = 0
+    for x in A
+        t = f(x)
+        if t ∉ seen
+            push!(seen,t)
+            count += 1
+            A[y[1]] = x
+            y = iterate(idxs, y[2])
+        end
+    end
+    resize!(A, count)
+end
+
 # If A is not grouped, then we will need to keep track of all of the elements that we have
 # seen so far.
 function _unique!(A::AbstractVector)
```

Comment:
This can be replaces now with

```
_unique!(A::AbstractVector) = unique(identity, A)
```
```

Reply:
@oxinabox Could you please explain it a little bit as this is my first contribution to Julia Lang?

Reply:
Sorry, I was being overly terse (unusual for me :-P) 

Once you have defined a function `unique!(f, A)`, like you are defining.
The function `_unique!(A)` here,
can be simply defined as call to `unique!(identity, A)`.

`identity` is a function in Julia that does `x->x`

Reply:
Okay, thanks for explaining it @oxinabox

Reply:
Modified _unique according to the need